pyvista/plotting/axes_assembly.py

```diff
@@ -303,9 +307,8 @@ def show_labels(self) -> bool:  # numpydoc ignore=RT01
     @show_labels.setter
     def show_labels(self, value: bool):  # numpydoc ignore=GL08
         self._show_labels = value
-        self._label_actors[0].SetVisibility(value)
-        self._label_actors[1].SetVisibility(value)
-        self._label_actors[2].SetVisibility(value)
+        for label in self._label_actor_iterator:
+            label.SetVisibility(value)
```

Comment:
This may be a different PR, but is it possible to set the value for each direction using a list?

Reply:
Do you mean you want to be able to control visibility of each label separately? Instead of all having the same visibility?

We can make the change. Probably best for a separate PR for the full feature. But, the changes in this PR make use of `self._label_actor_iterator` to abstract some of the label setter methods to the superclass, so that the method would apply to any/all labels, regardless if there are 3 (AxesAssembly) or 6 (AxesAssemblySymmetric). Because of this abstraction we lose fine-tuned control over each label actor.

Some options:

1. We could revert this abstraction, add a concrete implementation for the symmetric subclass, and modify the separate implementations of `show_labels`, `label_size`, etc. to allow setting the individual text properties.

2. An alternative, which is currently implemented in `AxesActor`, is to expose the separate actor props to the user, e.g. `AxesAssembly.x_label_prop`, `AxesAssembly.y_label_prop`, etc. But I'm personally not a fan of this, especially for the symmetric case since you need 6 lines of code to set all 6 labels. Plus there are `Text` attributes which we may to modify (e.g. visibility), but also `TextProperty` attributes, so this solution doesn't make it easy to set both kinds of attributes.

3. In the current implementation for #5029, I had added generic `set_prop` and `get_prop` methods (see the [code](https://github.com/pyvista/pyvista/blob/930fcfbf578ec3b45ac497199c94a21ea4ee1327/pyvista/plotting/axes_actor.py#L935-L1033)), allowing setting/getting actor properties on a part-basis (e.g. all shafts or all tips) or an axis basis (all x parts, all y parts, or all z parts). This was for the 3D actor properties, but we could add something like for the text label properties too, and do `try... except` on both the `Text` and `TextProperty` to make it simple to use.

Reply:
Just to add to this, options 2 and 3 aim to be generic to allow setting _any_ actor attribute, not just `size` or `visibility`. Here are some concrete examples... If we want to make the positive labels visibly but not the negative ones, for option 1 (your suggestion) this would maybe look like:
```python
assembly.show_labels = [True, False, True, False, True, False]
```
For option 3, we could keep the current implementation for simplicity, e.g. 
```python
assembly.show_labels = True
```
But then allow something like:
```python
assembly.set_label_prop('visibility', True, axis='positive')
```
This would also allow setting the justification, bold/italics, text shadow... etc., e.g. to center-justify the text for all labels:
```python
assembly.set_label_prop('justification_horizontal', 'center')
```

Reply:
Thank you. It was just a passing thought, so if it has that much of an impact, there's no need to do it. Let's leave it as it is and respond if there is a strong request.

Reply:
I now see the value in having this feature. If plotting with any plane view, it would be nice to hide the in-plane and out-of plane labels, e.g. hide `z` labels for `view_xy`. See the unsightly and unreadable stacked labels in the center of this widget:

<img width="103" alt="Screenshot 2024-08-15 at 12 52 39 AM" src="https://github.com/user-attachments/assets/7d0b7204-16bd-40ca-9628-4f3d6b524710">